recipes/cpprestsdk/all/conanfile.py

```diff
@@ -45,9 +47,9 @@ def _build_subfolder(self):
         return "build_subfolder"
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
+        files.copy(self, "CMakeLists.txt", src=self.recipe_folder, dst=self.export_sources_folder)
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+            files.copy(self, patch["patch_file"], src=self.recipe_folder, dst=self.export_sources_folder)
```

Comment:
You can replace all these lines `export_conandata_patches(self)`. It's a new tool available since Conan 1.52.0 which does the exactly same boiler plate. 

https://docs.conan.io/en/latest/reference/conanfile/tools/files/patches.html#conan-tools-files-export-conandata-patches